numba/compiler.py

```diff
@@ -706,18 +707,28 @@ def stage_objectmode_backend(self):
         lowerfn = self.backend_object_mode
         self._backend(lowerfn, objectmode=True)
 
-        # Warn if compiled function in object mode and force_pyobject not set
+        # Warn, deprecated behaviour, code compiled in objmode without
+        # force_pyobject indicates fallback from nopython mode
         if not self.flags.force_pyobject:
+            # first warn about object mode and yes/no to lifted loops
             if len(self.lifted) > 0:
                 warn_msg = ('Function "%s" was compiled in object mode without'
                             ' forceobj=True, but has lifted loops.' %
                             (self.func_id.func_name,))
             else:
                 warn_msg = ('Function "%s" was compiled in object mode without'
                             ' forceobj=True.' % (self.func_id.func_name,))
-            warnings.warn_explicit(warn_msg, errors.NumbaWarning,
-                                   self.func_id.filename,
-                                   self.func_id.firstlineno)
+            warnings.warn(errors.NumbaWarning(warn_msg,
+                                              self.func_ir.loc))
```

Comment:
This should be removed because there is another warning below, right?

Reply:
This is up for debate. There's a number of sites in the compiler pipeline which use warnings to tell users about, what is essentially, the state of the pipeline and why some part of it is failing. These have been present for a long time but not visible by default because of the warning filter in `numba.config`. Given the fall-back behaviour is being deprecated, perhaps it is sensible to also remove explanations/warnings of why the fall-back is occurring given they will be unreachable once the deprecation comes into force?

Reply:
Oh, I see.  OK, looking at the two messages, they are communicating different things.  I'm happy to leave this as-is then.

Reply:
Perhaps we should also check how noisy the test suite has become as a result of this?